src/effects/effectchainslot.cpp

```diff
@@ -165,102 +583,175 @@ void EffectChainSlot::slotChainChannelStatusChanged(const QString& group,
+        //         this, SLOT(slotChainChannelStatusChanged(const QString&, bool)));
 
         m_pControlChainLoaded->forceSet(true);
+
+        // NOTE(Kshitij) : Updated the following function
+        // m_pControlChainMixMode->set(
+        //         static_cast<double>(m_pEffectChain->mixMode()));
         m_pControlChainMixMode->set(
-                static_cast<double>(m_pEffectChain->mixMode()));
+                static_cast<double>(mixMode()));
 
         // Mix and enabled channels are persistent properties of the chain slot,
         // not of the chain. Propagate the current settings to the chain.
-        m_pEffectChain->setMix(m_pControlChainMix->get());
-        m_pEffectChain->setEnabled(m_pControlChainEnabled->get() > 0.0);
+
+        // NOTE(Kshitij) : Updated the functions
+        // m_pEffectChain->setMix(m_pControlChainMix->get());
+        // m_pEffectChain->setEnabled(m_pControlChainEnabled->get() > 0.0);
+        setMix(m_pControlChainMix->get());
+        setEnabled(m_pControlChainEnabled->get() > 0.0);
 
         // Don't emit because we will below.
         for (int i = 0; i < m_slots.size(); ++i) {
             slotChainEffectChanged(i, false);
         }
-    }
+    // }
 
-    emit(effectChainLoaded(pEffectChain));
+    // NOTE(Kshitij) : Not being used so redundant
+    // emit(effectChainLoaded(pEffectChain));
     emit(updated());
 }
 
 void EffectChainSlot::updateRoutingSwitches() {
-    VERIFY_OR_DEBUG_ASSERT(m_pEffectChain) {
+    // NOTE(Kshitij) : Updated the assert line to return statement
+    // VERIFY_OR_DEBUG_ASSERT(m_pEffectChain) {
+    //     return;
+    // }
+    
+    // NOTE(Kshitij) : Removed m_pEffectChain dependancy
+    // if (!m_pEffectChain) {
+    //     return;
+    // }
+    // for (const ChannelInfo* pChannelInfo : m_channelInfoByName) {
+    //     if (pChannelInfo->pEnabled->toBool()) {
+    //         m_pEffectChain->enableForInputChannel(pChannelInfo->handle_group);
+    //     } else {
+    //         m_pEffectChain->disableForInputChannel(pChannelInfo->handle_group);
+    //     }
+    // }
+    qDebug() << "updatingRoutingSwitches()";
+    if (isEmpty()) {
         return;
     }
     for (const ChannelInfo* pChannelInfo : m_channelInfoByName) {
         if (pChannelInfo->pEnabled->toBool()) {
-            m_pEffectChain->enableForInputChannel(pChannelInfo->handle_group);
+            enableForInputChannel(pChannelInfo->handle_group);
         } else {
-            m_pEffectChain->disableForInputChannel(pChannelInfo->handle_group);
+            disableForInputChannel(pChannelInfo->handle_group);
         }
     }
 }
 
 EffectChainPointer EffectChainSlot::getEffectChain() const {
+    // TODO(Kshitij) : Remove this function
     return m_pEffectChain;
 }
 
+// TODO(Kshitij) : Return EffectChainPreset instead of EffectChain
+// IMPORTANT(Kshitij) : Check what to return and keep the code working
 EffectChainPointer EffectChainSlot::getOrCreateEffectChain(
```

Comment:
This function was another ugly piece of duct tape to keep the old architecture from crumbling. Let's remove it.